sdcflows/transform.py

```diff
@@ -72,151 +274,271 @@ def fit(self, spatialimage):
 
         """
         # Calculate the physical coordinates of target grid
-        if isinstance(spatialimage, (str, bytes, Path)):
-            spatialimage = nb.load(spatialimage)
+        if isinstance(target_reference, (str, bytes, Path)):
+            target_reference = nb.load(target_reference)
+
+        approx &= xfm_data2fmap is not None  # Approximate iff xfm_data2fmap is defined
+        xfm_data2fmap = xfm_data2fmap if xfm_data2fmap is not None else np.eye(4)
+        target_affine = target_reference.affine.copy()
+        target_header = target_reference.header.copy()
+
+        if approx:
+            _tmp_shape = target_reference.shape[:3]
+
+        # Project the reference's grid onto the fieldmap's
+        target_reference = target_reference.__class__(
+            target_reference.dataobj,
+            xfm_data2fmap @ target_affine,
+            target_header,
+        )
```

Comment:
This feels hard to follow. We use `target_affine` and `target_header` late on to get back to the original reference. What if we create a new image instead:

```suggestion
        # Project the reference's grid onto the fieldmap's
        projected_reference = target_reference.__class__(
            target_reference.dataobj,
            xfm_data2fmap @ target_reference.affine,
            target_reference.header,
        )
```

This would make it easier to keep track of which "target reference" is intended at all points.